noir-projects/noir-contracts/contracts/private_fpc_contract/src/main.nr

```diff
@@ -1,35 +1,39 @@
-mod lib;
+mod settings;
 
 contract PrivateFPC {
-    use dep::aztec::{protocol_types::{address::AztecAddress, hash::poseidon2_hash}, state_vars::SharedImmutable};
+    use dep::aztec::{protocol_types::{address::AztecAddress, hash::compute_siloed_nullifier}, state_vars::SharedImmutable};
     use dep::token_with_refunds::TokenWithRefunds;
-    use crate::lib::emit_randomness_as_unencrypted_log;
+    use crate::settings::Settings;
 
     #[aztec(storage)]
     struct Storage {
-        other_asset: SharedImmutable<AztecAddress>,
-        admin: SharedImmutable<AztecAddress>,
+        settings: SharedImmutable<Settings>,
```

Comment:
Did this change as it will allow us to drop the gate count by 4k once we optimize SharedImmutable to perfom only 1 merkle proof.